tests/test_model.py

```diff
@@ -0,0 +1,636 @@
+    """Test TeslaFiVehicle class id property"""
+    vehicle_data = {
+        "vehicle_id": "FOO_VEHICLE_ID",
+        "vin": "5YJSA1CG3DFP14555",
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.vehicle_id == "FOO_VEHICLE_ID"
+
+def test_TeslaFiVehicle_vehicle_id_use_vin():
+    """Test TeslaFiVehicle class id property"""
+    vehicle_data = {
+        "vin": "5YJSA1CG3DFP14555"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.vehicle_id == "5YJSA1CG3DFP14555"
+
+def test_TeslaFiVehicle_firmware_version():
+    """Test TeslaFiVehicle class id property"""
+    vehicle_data = {
+        "car_version": "2025.2.9"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.firmware_version == "2025.2.9"
+
+def test_TeslaFiVehicle_firmware_version_none():
+    """Test TeslaFiVehicle class firmware_version property"""
+    vehicle_data = {
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.firmware_version is None
+
+def test_TeslaFiVehicle_name():
+    """Test TeslaFiVehicle class name property"""
+    vehicle_data = {
+        "display_name": "Model 3"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.name == "Model 3"
+
+def test_TeslaFiVehicle_name_none():
+    """Test TeslaFiVehicle class name property"""
+    vehicle_data = {
+        "vin": "5YJSA1CG3DFP14555",
+        "display_name": None
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.name == "2013 modelS 4555"
+    
+def test_TeslaFiVehicle_last_remote_update():
+    """Test TeslaFiVehicle class last_remote_update property"""
+    vehicle_data = {
+        "Date": "2023-10-01 12:00:00"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.last_remote_update == datetime.strptime("2023-10-01 12:00:00", TESLAFI_DATE_FORMAT)
+
+def test_TeslaFiVehicle_last_remote_update_none():
+    """Test TeslaFiVehicle class last_remote_update property"""
+    vehicle_data = {
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    assert vehicle.last_remote_update is None
+
+def test_TeslaFiVehicle_last_remote_update_parse_error():
+    """Test TeslaFiVehicle class last_remote_update property"""
+    vehicle_data = {
+        "Date": "XXXX-10-01 12:00:00"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    assert vehicle.last_remote_update is None
+
+def test_TeslaFiVehicle_car_type():
+    """Test TeslaFiVehicle class car_type property"""
+    vehicle_data = {
+        "car_type": "modelS"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.car_type == "modelS"
+
+def test_TeslaFiVehicle_car_type_from_vin():
+    """Test TeslaFiVehicle class car_type property"""
+    vehicle_data = {
+        "car_type": "modelS"
+    }
+    
+    vehicle = TeslaFiVehicle(vehicle_data)
+    
+    assert vehicle.car_type == "modelS"
```

Comment:
Looks like this is duplicating `test_TeslaFiVehicle_car_type`

```suggestion
def test_TeslaFiVehicle_car_type_from_vin():
    """Test TeslaFiVehicle class car_type property"""
    vehicle_data = {
        "vin": _TEST_VIN
    }
    
    vehicle = TeslaFiVehicle(vehicle_data)

    assert vehicle.car_type == "modelS"
```